### PR TITLE
[FSTORE-581] Update fv doc string

### PR DESCRIPTION
**Pull request by kennethmhc**
This PR adds/fixes/changes...
- please summarize your changes to the code 
- and make sure to include all changes to user-facing APIs

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [ ] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/feature_view.py [hunk 1]
```diff
@@ -109,16 +109,12 @@ def delete(self):
 
     @staticmethod
     def clean(feature_store_id: int, feature_view_name: str, feature_view_version: str):
-        """Delete the feature view and all associated metadata.
+        """
+        Delete the feature view and all associated metadata and training data.
+        This can delete corrupted feature view which cannot be retrieved due to e.g corrupted query.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:114]:
```suggestion
        This can delete corrupted a feature view which cannot be retrieved due to a corrupted query for example.
```

File: python/hsfs/feature_view.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def __init__(
         self._batch_scoring_server = None
 
     def delete(self):
-        """Delete current feature view and all associated metadata.
+        """Delete current feature view and all associated metadata and training data.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:81]:
```suggestion
        """Delete current feature view, all associated metadata and training data.
```

File: python/hsfs/feature_view.py [hunk 3]
```diff
@@ -187,15 +182,13 @@ def init_serving(
...
             feature_view = fs.get_feature_view(...)
 
-            # initialise and cache parametrized prepared statement to retrieve a feature vector
+            # initialise feature view to retrieve a feature vector
             feature_view.init_serving(training_dataset_version=1)
             ```
 
         # Arguments
             training_dataset_version: int, optional. Default to be 1. Transformation statistics
                 are fetched from training dataset and apply in serving vector.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:191]:
```suggestion
                are fetched from training dataset and applied to the serving vector.
```

Comment (moritzmeister) [reply python/hsfs/feature_view.py:191]: same for the other method

File: python/hsfs/feature_view.py [hunk 4]
```diff
@@ -589,7 +582,8 @@ def create_training_data(
         statistics_config: Optional[Union[StatisticsConfig, bool, dict]] = None,
         write_options: Optional[Dict[Any, Any]] = {},
     ):
-        """Create a training dataset and save data into `location`.
+        """Create a training dataset metadata and save the training data into `location`.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:585]:
```suggestion
        """Create the metadata for a training dataset and save the corresponding training data into `location`.
```

File: python/hsfs/feature_view.py [hunk 5]
```diff
@@ -589,7 +582,8 @@ def create_training_data(
...
+        The training data can be retrieved by `feature_view.get_training_data`.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:586]:
```suggestion
        The training data can be retrieved by calling `feature_view.get_training_data`.
```

File: python/hsfs/feature_view.py [hunk 6]
```diff
@@ -786,7 +780,9 @@ def create_train_test_split(
         statistics_config: Optional[Union[StatisticsConfig, bool, dict]] = None,
         write_options: Optional[Dict[Any, Any]] = {},
     ):
-        """Create a training dataset and save data into `location`.
+        """Create a training dataset metadata and save the training data into `location`.
+        The training data is splited into train and test set at random or according to time ranges.
+        The training data can be retrieved by `feature_view.get_train_test_split`.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:785]:
```suggestion
        """Create the metadata for a training dataset and save the corresponding training data into `location`.
        The training data is split into train and test set at random or according to time ranges.
        The training data can be retrieved by calling `feature_view.get_train_test_split`.
```

File: python/hsfs/feature_view.py [hunk 7]
```diff
@@ -1040,6 +1036,8 @@ def create_train_validation_test_split(
         write_options: Optional[Dict[Any, Any]] = {},
     ):
         """Create a training dataset and save data into `location`.
+        The training data is splited into train, validation, and test set at random or according to time range.
+        The training data can be retrieved by `feature_view.get_train_validation_test_split`.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:1040]:
```suggestion
        """Create the metadata for a training dataset and save the corresponding training data into `location`.
        The training data is split into train, validation, and test set at random or according to time range.
        The training data can be retrieved by calling `feature_view.get_train_validation_test_split`.
```

File: python/hsfs/feature_view.py [hunk 8]
```diff
@@ -1318,7 +1316,9 @@ def training_data(
         read_options: Optional[Dict[Any, Any]] = None,
     ):
         """
-        Get training data from feature groups.
+        Create a training dataset metadata and get training data from the offline feature store.
+        This returns the training data in memory and does not materialise data in storage.
+        The training data can be recreated by `feature_view.get_training_data` with the metadata created.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:1321]:
```suggestion
        Create the metadata for a training dataset and get the corresponding training data from the offline feature store.
        This returns the training data in memory and does not materialise data in storage.
        The training data can be recreated by calling `feature_view.get_training_data` with the metadata created.
```

File: python/hsfs/feature_view.py [hunk 9]
```diff
@@ -1423,7 +1423,10 @@ def train_test_split(
         read_options: Optional[Dict[Any, Any]] = None,
     ):
         """
-        Get training data from feature groups.
+        Create a training dataset metadata and get training data from the offline feature store.
+        This returns the training data in memory and does not materialise data in storage.
+        The training data is splited into train and test set at random or according to time ranges.
+        The training data can be recreated by `feature_view.get_train_test_split` with the metadata created.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:1429]:
```suggestion
        Create the metadata for a training dataset and get the corresponding training data from the offline feature store.
        This returns the training data in memory and does not materialise data in storage.
        The training data is split into train and test set at random or according to time ranges.
        The training data can be recreated by calling `feature_view.get_train_test_split` with the metadata created.
```

File: python/hsfs/feature_view.py [hunk 10]
```diff
@@ -1560,7 +1563,10 @@ def train_validation_test_split(
         read_options: Optional[Dict[Any, Any]] = None,
     ):
         """
-        Get training data from feature groups.
+        Create a training dataset metadata and get training data from the offline feature store.
+        This returns the training data in memory and does not materialise data in storage.
+        The training data is splited into train, validation, and test set at random or according to time ranges.
+        The training data can be recreated by `feature_view.get_train_validation_test_split` with the metadata created.
```
Comment (moritzmeister) [inline python/hsfs/feature_view.py:1569]:
```suggestion
        Create the metadata for a training dataset and get the corresponding training data from the offline feature store.
        This returns the training data in memory and does not materialise data in storage.
        The training data is split into train, validation, and test set at random or according to time ranges.
        The training data can be recreated by calling `feature_view.get_train_validation_test_split` with the metadata created.
```

**Review by moritzmeister — CHANGES_REQUESTED** [review]

**Review by moritzmeister — APPROVED** [review]
